### PR TITLE
PEPPER-1105 New Playwright test for Onc History Facility lookup

### PR DESCRIPTION
**Pull request by aweng98**
[PEPPER-1105](https://broadworkbench.atlassian.net/browse/PEPPER-1105)

Test run pass in Dev and Test env.

Screenshot 2024-02-15 at 9 49 57 AM

<hr>

[video (1).webm](https://github.com/broadinstitute/ddp-angular/assets/35533885/e63398de-06ff-4d75-8fe1-7c8d7e290afa)

[PEPPER-1105]: https://broadworkbench.atlassian.net/browse/PEPPER-1105?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: playwright-e2e/tests/dsm/onc-history/pancan-onc-history-facility-lookup.spec.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          {
+            date: {
+              date: {
+                yyyy: new Date().getFullYear(),
+                month: new Date().getMonth(),
+                dayOfMonth: new Date().getDate()
+              }
+            }
+          }, lastRow);
+      await oncHistoryTable.fillField(OncHistoryInputColumnsEnum.FACILITY, { value: newFacilityName, lookupSelectIndex: -1 }, lastRow);
```
Comment (aweng98) [inline playwright-e2e/tests/dsm/onc-history/pancan-onc-history-facility-lookup.spec.ts:72]: Set `lookupSelectIndex: -1` to skip facility lookup.

File: playwright-e2e/dsm/component/tables/onc-history-table.ts [hunk 2]
```diff
@@ -128,30 +128,34 @@ export default class OncHistoryTable extends Table {
...
     if (currentValue !== actualValue) {
-      await Promise.all([
-        waitForResponse(this.page, { uri: 'patch' }),
-        inputElement.fillSimple(actualValue)
+      const [resp] = await Promise.all([
+        waitForResponse(this.page, { uri: '/patch' }),
+        input.fillSimple(actualValue)
       ]);
-      hasLookup && await this.lookup(lookupSelectIndex, true);
+      if (hasLookup && lookupSelectIndex >= 0) {
```
Comment (aweng98) [inline playwright-e2e/dsm/component/tables/onc-history-table.ts:145]: Adding `lookupSelectIndex >= 0` check for skip using Lookup.

File: playwright-e2e/dsm/component/tables/onc-history-table.ts [hunk 3]
```diff
@@ -196,28 +202,31 @@ export default class OncHistoryTable extends Table {
...
         selectInput.selectOption(selectRequest)
       ]);
       return resp;
     }
     return null;
   }
 
-  private async lookup(selectIndex = 0, isFacility = false): Promise<void> {
-    const lookupList = this.lookupList;
+  private async lookup(root: Locator, selectIndex = 0, isFacility = false): Promise<string> {
```
Comment (aweng98) [inline playwright-e2e/dsm/component/tables/onc-history-table.ts:213]: `root` param narrows down the area of search for the Lookup.

**Comment by aweng98** [comment]
Passed in [CircleCI](https://app.circleci.com/pipelines/github/broadinstitute/ddp-angular/27054/workflows/0fa5244a-855e-4e88-a3a4-4c85b0d3bd33).

File: playwright-e2e/dsm/component/tables/onc-history-table.ts [hunk 4]
```diff
@@ -294,15 +303,15 @@ export default class OncHistoryTable extends Table {
...
+    return `//app-onc-history-detail//table[contains(@class,"table")]`
   }
 
   private get rowXPath(): string {
     return '/tbody/tr'
   }
 
   private get lookupListXPath(): string {
-    return '//app-lookup/div/ul/li'
+    return '//ancestor-or-self::td//app-lookup/div/ul/li'
```
Comment (aweng98) [inline playwright-e2e/dsm/component/tables/onc-history-table.ts:314]: The Lookup in an input or textarea field should have the same `td` as the root element.

**Review by Rinchan20 — APPROVED** [review]

**Pull request merged**